src/node.c

```diff
@@ -30,9 +30,6 @@ static bool S_can_contain(cmark_node *node, cmark_node *child) {
   if (node == NULL || child == NULL) {
     return false;
   }
-  if (NODE_MEM(node) != NODE_MEM(child)) {
```

Comment:
What if, instead of enforcing this constraint in `can_contain`, we simply left it up to the users of the library not to shoot themselves in the foot by using different allocators for parents and children? It seems to me that it would be pretty rare for someone to do this, anyway, and I'm worried that including this check in `can_contain` is making perfectly normal and legitimate uses of the library difficult or impossible.